File: extensions/s3eFlurry/source/android/s3eFlurry.java
/*
java implementation of the s3eFlurry extension.

Add android-specific functionality here.

These functions are called via JNI from native code.
*/
/*
 * NOTE: This file was originally written by the extension builder, but will not
 * be overwritten (unless --force is specified) and is intended to be modified.
 */
import com.ideaworks3d.marmalade.LoaderAPI;
import com.ideaworks3d.marmalade.LoaderActivity;

import android.view.View;
import android.view.ViewGroup;

import com.flurry.android.*;

import android.util.Log;



//import android.content.Context;
import android.R;
import android.widget.Button;
import android.widget.TextView;
import android.graphics.Typeface;
import android.view.Gravity;
import android.widget.LinearLayout;
import android.widget.FrameLayout;

class s3eFlurry
{
    AppCircle appCircle;
    View m_PromoView;
    boolean m_isBannerDisplayed = false;
    String message = "Flurry for Marmalade SDK";

    // Flurry Analaytics
    public void s3eFlurryStart(String ID)
    {
        FlurryAgent.onStartSession(LoaderAPI.getActivity(), ID);
        Log.d("s3eFlurry", "////// App Circle Session Started //////");
    }
    public void s3eFlurryLogEvent(String eventName, boolean timed)
    {
        if(!timed)
            FlurryAgent.logEvent(eventName);
        else
            FlurryAgent.logEvent(eventName, timed);
    }
    public void s3eFlurryEndTimedEvent(String eventName)
    {
        FlurryAgent.endTimedEvent(eventName);
    }
    public void s3eFlurryLogError(String errorName, String errorMessage)
    {
        FlurryAgent.onError(errorName, errorMessage, "");
    }
    public void s3eFlurrySetUserID(String userID)
    {
        FlurryAgent.setUserId(userID);
    }
    public void s3eFlurrySetUserAge(int age)
    {
        FlurryAgent.setAge(age);
    }
    public void s3eFlurrySetUserGender(boolean isMale)
    {
        if(isMale)
            FlurryAgent.setGender(Constants.MALE);
        else
            FlurryAgent.setGender(Constants.FEMALE);
    }
    public void s3eFlurrySetLocation()
    {
    }
    public void s3eFlurrySetSessionReportOnClose()
    {
    }
    public void s3eFlurrySetSessionReportOnPause()
    {
    }

    // Flurry App Circle
    public void s3eFlurryAppCircleEnable()
    {
        FlurryAgent.enableAppCircle();
        Log.d("s3eFlurry", "////// App Circle Enabled //////");
    }
    public void s3eFlurrySetDefaultText(String text)
    {
        Log.d("S3EFLURRY", "Setting text");
        message = text;
        Log.d("s3eFlurry", "////// Default Banner Text Changed: " + message + " //////");
    }
    public void s3eFlurryShowAdBanner(boolean showBanner)
    {
        if(true == showBanner) // Show Banner
        {
            appCircle = FlurryAgent.getAppCircle();
            if(appCircle != null)
            {
                appCircle.setDefaultNoAdsMessage(message);

                final ViewGroup viewGroup = LoaderActivity.m_Activity.GetFrameLayout();
                m_PromoView = appCircle.getHook(LoaderActivity.m_Activity, "s3eAPPCIRCLE_BANNER_HOOK", com.flurry.android.Constants.MODE_LANDSCAPE);

                if (viewGroup == null)
                {
                    Log.d("View Error", "View group not found");
                    return;
                }

                if (m_PromoView == null)
                {
                    Log.d("Flurry Banner Error", "Banner View Failed");
                    return;
                }

                // Define the layout parameters of the view
                final LinearLayout.LayoutParams lp = new LinearLayout.LayoutParams(
                                                        LinearLayout.LayoutParams.FILL_PARENT,//LinearLayout.LayoutParams.WRAP_CONTENT,
                                                        80);//LinearLayout.LayoutParams.FILL_PARENT);

                LoaderAPI.getActivity().LoaderThread().runOnOSThread(new Runnable()
                {
                    public void run()
                    {
                        if(m_PromoView != null && !m_isBannerDisplayed)
                        {
                            // Must add our group to index level 1 or it will not be shown
                            viewGroup.addView(m_PromoView, 1, lp);
                            Log.d("s3eFlurry", "Banner Displayed");
                            m_isBannerDisplayed = true; // We should ensure we only add the view once
                        }
                    }
                });
            }
            else
                Log.d("S3EFLURRY", "App Circle is not enabled");
        }
        else // Hide Banner
        {
            final ViewGroup viewGroup = LoaderActivity.m_Activity.GetFrameLayout();
            viewGroup.removeView(m_PromoView);
            Log.d("s3eFlurry", "////// View Removed //////");
            m_isBannerDisplayed = false;
        }
    }
}

File: extensions/s3eFlurry/source/generic/s3eFlurry.cpp
/*
Generic implementation of the s3eFlurry extension.
This file should perform any platform-indepedentent functionality
(e.g. error checking) before calling platform-dependent implementations.
*/

/*
 * NOTE: This file was originally written by the extension builder, but will not
 * be overwritten (unless --force is specified) and is intended to be modified.
 */


#include "s3eFlurry_internal.h"
s3eResult s3eFlurryInit()
{
    //Add any generic initialisation code here
    return s3eFlurryInit_platform();
}

void s3eFlurryTerminate()
{
    //Add any generic termination code here
    s3eFlurryTerminate_platform();
}

void s3eFlurryStart(const char* ID)
{
	s3eFlurryStart_platform(ID);
}

void s3eFlurryLogEvent(const char* eventName, const s3eBool timed)
{
	s3eFlurryLogEvent_platform(eventName, timed);
}

void s3eFlurryEndTimedEvent(const char* eventName)
{
	s3eFlurryEndTimedEvent_platform(eventName);
}

void s3eFlurryLogError(const char* errorName, const char* errorMessage)
{
	s3eFlurryLogError_platform(errorName, errorMessage);
}

void s3eFlurrySetUserID(const char* userID)
{
	s3eFlurrySetUserID_platform(userID);
}

void s3eFlurrySetUserAge(const uint8 age)
{
	s3eFlurrySetUserAge_platform(age);
}

void s3eFlurrySetUserGender(const s3eFlurryUserGender gender)
{
	s3eFlurrySetUserGender_platform(gender);
}

void s3eFlurrySetLocation(s3eLocation* location)
{
	s3eFlurrySetLocation_platform(location);
}

void s3eFlurrySetSessionReportOnClose(const s3eBool sendReportOnClose)
{
	s3eFlurrySetSessionReportOnClose_platform(sendReportOnClose);
}

void s3eFlurrySetSessionReportOnPause(const s3eBool sendReportOnPause)
{
	s3eFlurrySetSessionReportOnPause_platform(sendReportOnPause);
}

void s3eFlurryAppCircleEnable()
{
	s3eFlurryAppCircleEnable_platform();
}

void s3eFlurrySetDefaultText(const char* text)
{
	s3eFlurrySetDefaultText_platform(text);
}

void s3eFlurryShowAdBanner(const s3eBool show)
{
	s3eFlurryShowAdBanner_platform(show);
}

void s3eFlurryShowOfferWall()
{
	s3eFlurryShowOfferWall_platform();
}
